spec/unit/facter/java_version_spec.rb

```diff
@@ -6,22 +6,48 @@
   }
 
   describe "java_version" do
+    context 'returns java version when java present on OpenBSD' do
+      before do
+        Facter.fact(:osfamily).stubs(:value).returns("OpenBSD")
+      end
+      let(:facts) { {:osfamily => 'OpenBSD'} }
+      it do
+        java_version_output = <<-EOS1
+openjdk version "1.7.0_71"
+OpenJDK Runtime Environment (build 1.7.0_71-b14)
+OpenJDK 64-Bit Server VM (build 24.71-b01, mixed mode)
+        EOS1
+        Facter::Util::Resolution.expects(:which).with("/usr/local/jdk-1.7.0/jre/bin/java").returns(true)
+        Facter::Util::Resolution.expects(:exec).with("/usr/local/jdk-1.7.0/jre/bin/java -Xmx8m -version 2>&1").returns(java_version_output)
+        Facter.value(:java_version).should == "1.7.0_71"
+      end
+    end
     context 'returns java version when java present' do
       it do
-        java_version_output = <<-EOS
+        java_version_output = <<-EOS2
```

Comment:
You don't need to have different heredoc codes so long as they don't overlap, so this change is unnecessary.